module/zfs/vdev.c

```diff
@@ -892,7 +892,7 @@ vdev_metaslab_init(vdev_t *vd, uint64_t txg)
 
 	ASSERT(oldc <= newc);
 
-	mspp = kmem_zalloc(newc * sizeof (*mspp), KM_SLEEP);
+	mspp = vmem_zalloc(newc * sizeof (*mspp), KM_SLEEP);
```

Comment:
Yes, you're right. I probably didn't do that because it happens to be the same under the hood. But there's no guarantee it won't change.